onnxruntime/core/providers/stvm/stvm_execution_provider.cc

```diff
@@ -0,0 +1,538 @@
+              } }, true);
+  }
+
+  auto meta_def = std::make_unique<::onnxruntime::IndexedSubGraph::MetaDef>();
+  meta_def->name = "TVMStandalone";
+  meta_def->domain = "StandaloneTest";
+  std::vector<std::string> inputs;
+  std::vector<std::string> outputs;
+
+  for (auto& nodeArgPtr : graph_viewer.GetInputs()) {
+    inputs.push_back(nodeArgPtr->Name());
+  }
+
+  for (auto& name : required_initializers) {
+    inputs.push_back(name);
+  }
+
+  for (auto& nodeArgPtr : graph_viewer.GetOutputs()) {
+    outputs.push_back(nodeArgPtr->Name());
+  }
+  meta_def->inputs = inputs;
+  meta_def->outputs = outputs;
+  meta_def->since_version = 1;
+  meta_def->status = ONNX_NAMESPACE::EXPERIMENTAL;
+  sub_graph->SetMetaDef(std::move(meta_def));
+  sub_graph->nodes = sorted_nodes;
+  result.push_back(
+      std::make_unique<ComputeCapability>(std::move(sub_graph)));
+  return result;
+}
+
+common::Status StvmExecutionProvider::Compile(const std::vector<Node*>& nodes,
+                                              std::vector<NodeComputeInfo>& node_compute_funcs) {
+  for (auto* fused_node : nodes) {
+    auto func_body = fused_node->GetFunctionBody();
+    if (!func_body)
+      return common::Status(common::ONNXRUNTIME, common::INVALID_ARGUMENT, "Function body is empty");
+    const std::string func_name = fused_node->Name();
+    const Graph& node_graph = func_body->Body();
+    Model model(node_graph.Name(), true, ModelMetaData(), PathString(),
+                             IOnnxRuntimeOpSchemaRegistryList(), node_graph.DomainToVersionMap(),
+                             std::vector<ONNX_NAMESPACE::FunctionProto>(), *GetLogger());
+    ONNX_NAMESPACE::ModelProto model_proto = model.ToProto();
+
+    model_proto.set_ir_version(ONNX_NAMESPACE::Version::IR_VERSION);
+    *(model_proto.mutable_graph()) = node_graph.ToGraphProto();
+    auto opset = model_proto.add_opset_import();
+    opset->set_domain(kOnnxDomain);
+    opset->set_version(node_graph.DomainToVersionMap().at(kOnnxDomain));
+
+    std::string string_buf;
+    model_proto.SerializeToString(&string_buf);
+    buffers_[func_name] = string_buf;
+    opsets_[func_name] = int(opset->version());
+    model_paths_[func_name] = fused_node->ModelPath().ToPathString();;
+
+    if (dump_subgraphs_) {
+        std::fstream dump("/tmp/" + fused_node->Name() + ".onnx", std::ios::out | std::ios::trunc | std::ios::binary);
+        model_proto.SerializeToOstream(&dump);
+    }
+
+    NodeComputeInfo compute_info;
+    compute_info.create_state_func = std::bind(&StvmExecutionProvider::CreateStateFunc, this, std::placeholders::_1, std::placeholders::_2);
+
+    compute_info.release_state_func = [](FunctionState state) {
+      if (state)
+        delete static_cast<STVMFuncState*>(state);
+    };
+
+    runners_[func_name] = std::make_shared<STVMRunner>(this, func_name, node_graph);
+    compute_info.compute_func = *runners_[func_name].get();
```

Comment:
What happens if the input model contains ops that TVM cannot handle? (Eg., custom ops.) Where does it get handled? Most of the code here seems to grab the whole graph, without checking which ops can be handled. I assume some stage of the compilation will produce an error if it encounters an op it cannot handle?

Reply:
Check of ops occurs during compilation of a model inside TVM. If TVM does not support any op (e.g. customized) the error will be caught and creating of InferenceSession fails.

Reply:
IMO, the right strategy would be for the TVM EP to gracefully pass the responsibility to other EPs, like the default provider (if it is unable to handle any op). I think it is okay to add such functionality in a future PR. It may help at add a TODO comment as a reminder.

Reply:
Agreed @gramalingam; a TODO would be appropriate here.

Reply:
Thanks! I added TODO, we need time to investigate the mechanism of gracefully pass the responsibility to other EPs